src/content/containers/Root.js

```diff
@@ -28,26 +57,31 @@ class ProfileViewWhenReadyImpl extends Component {
 
   render() {
     const { view, dataSource } = this.props;
-    switch (view) {
+    switch (view.step) {
       case 'INITIALIZING': {
-        switch (dataSource) {
-          case 'none':
-            return <Home />;
-          case 'from-addon':
-            return <div>Retrieving profile from the gecko profiler addon...</div>;
-          case 'from-file':
-            return <div>Reading the file and parsing the profile in it...</div>;
-          case 'local':
-            return <div>Not implemented yet.</div>;
-          case 'public':
-            return <div>Retrieving profile from the public profile store...</div>;
-          default:
-            return <div>View not found.</div>;
+        if (dataSource === 'none') {
+          return <Home />;
+        }
+
+        const message = LOADING_MESSAGES[dataSource] || 'View not found';
+        let additionalMessage = null;
+        if (view.additionalData && view.additionalData.attempt) {
+          const attempt = view.additionalData.attempt;
+          additionalMessage = `Tried ${fewTimes(attempt.count)} out of ${attempt.total}.`;
         }
+
+        return (
+          <div>
+            <div>{ message }</div>
+            { additionalMessage && <div>{ additionalMessage }</div>}
+          </div>
+        );
       }
+      case 'FATAL_ERROR':
+        return <div>Couldn't load the profile from the store.</div>;
```

Comment:
`<div>{"Couldn't load the profile from the store."}</div>` will probably transpile to the same code.